OpenSim/Common/ComponentList.h

```diff
@@ -76,6 +78,21 @@ class ComponentFilterMatchAll : public ComponentFilter {
         return new ComponentFilterMatchAll(*this);
     }
 };
+
+/** A component is considered a match if its full path name matches the 
+provided regular expression. */
+class ComponentFilterFullPathNameRegex : public ComponentFilter {
+public:
+    ComponentFilterFullPathNameRegex(const std::regex& expr) : _expr(expr) {}
```

Comment:
Maybe accept a `string` rather than reference to `regex` itself. That way, user of this class need not create an object of type `regex` to use this class. It also avoids copying of `regex` object. They can just pass a string and internally, this class can either:
* Create a `regex` and keep it as private member.
* Create a use a `regex` on the fly having a `string` as private member.